src/ops/fuelup_toolchain/export.rs

```diff
@@ -0,0 +1,98 @@
+use crate::{
+    commands::toolchain::ExportCommand,
+    toolchain::Toolchain,
+    constants::FUEL_TOOLCHAIN_TOML_FILE,
+    toolchain_override::{self, OverrideCfg, ToolchainCfg, ToolchainOverride},
+};
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::str::FromStr;
+use semver::Version;
+use anyhow::{bail, Result};
+use component::{self, Components};
+use tracing::info;
+
+pub fn export(command: ExportCommand) -> Result<()> {
+    let ExportCommand { name, force} = command;
+    let path = PathBuf::from("./").join(FUEL_TOOLCHAIN_TOML_FILE);
+    if !force && path.exists() {
+        bail!("{FUEL_TOOLCHAIN_TOML_FILE} already exists");
+    }
+    let mut toolchain_name = Toolchain::from_settings()?.name;
+    if let Some(name) = name {
+        toolchain_name = name;
+    }
+    let export_toolchain = Toolchain::from_path(&toolchain_name);
+
+    let mut version_map: HashMap<String, Version> = HashMap::new();
+    for component in Components::collect_exclude_plugins()? {
+        let component_executable = export_toolchain.bin_path.join(&component.name);
+        if let Ok(version) = get_exec_version(component_executable.as_path()) {
+            version_map.insert(component.name.clone(), version);
+        };
+
+        if component.name == component::FORC {
+            let forc_executables = component.executables;
+            for plugin in Components::collect_plugins()? {
+                if !forc_executables.contains(&plugin.name) {
+                    if !plugin.is_main_executable() {
+                        for executable in plugin.executables.iter() {
+                            let plugin_executable = export_toolchain.bin_path.join(executable);
+                            if let Ok(version) = get_exec_version(plugin_executable.as_path()) {
+                                version_map.insert(executable.clone(), version);
+                            };
+                        }
+                    } else {
+                        let plugin_executable = export_toolchain.bin_path.join(&plugin.name);
+                        if let Ok(version) = get_exec_version(plugin_executable.as_path()) {
+                            version_map.insert(plugin.name.clone(), version);
+                        };
+                    }
+                }
+            }
+        }
+    }
+
+    let toolchain_override = ToolchainOverride {
+        cfg: OverrideCfg::new(
+            ToolchainCfg {
+                channel: toolchain_override::Channel::from_str(&toolchain_name).unwrap(),
```

Comment:
I don't think this will work for custom toolchains. The channel must be one of the distributed channels; it can't be an arbitrary name of a custom toolchain. See `deserialize_channel`

Some options are: 
1. We could default the channel to latest. But, that's making an assumption.
2. We can ask the user to provide a valid channel and then continue the export.
3. We can restrict users to only be able to export their toolchain if they're using a distributed toolchain. This would still be useful for just generating a fuel-toolchain.toml file, as `forc init` currently does not generate this file.
4. In the case where a user supplies `--force` and the file already exists, we could use the `channel` from the previous file, and only override the components where they differ from the user's default toolchain. 

I don't have a strong opinion on this, but I'd lean towards doing 2 & 4.